x/deployment/keeper/grpc_query_test.go

```diff
@@ -149,15 +149,15 @@ func TestGRPCQueryDeployments(t *testing.T) {
 				req = &types.QueryDeploymentsRequest{
 					Filters: types.DeploymentFilters{
 						DSeq:  37,
-						State: types.Deployment_State(4),
+						State: "closed",
```

Comment:
why did you change the type to a string?

Reply:
In cli, state filter is taken as `string` and converted to `int`. But we can't do this conversion in grpc rest as there is no way to do it. To solve this, we changed filter type to `string` and converting state value to int in grpc querier instead of in cli.

Reply:
thanks.  you mean in grpc-gateway?  can you show me where?  Ideally the api would accept proto generated types only.

Reply:
Yes, api would accept proto generated types. That's the reason I changed `State` in `DeploymentFilters` to string [here](https://github.com/ovrclk/akash/pull/803/files#diff-4fbe7df455aab18e9713e6819c2adf53R149),

And I am converting state filter to int [here](https://github.com/ovrclk/akash/pull/803/files#diff-80d646c61a94fd1837b53588833bb71bR28-R34) instead of doing it in cli [previously](https://github.com/ovrclk/akash/pull/803/files#diff-dcdc703455bc99749c6ba962ddb6d7edL51-L56)

Reply:
what is grpc rest?

Reply:
grpc rest is the rest client of grpc which uses grpc gateway

Reply:
@boz gRPC was using the original deployment filter types (i.e., `int` values for `status` filters) but that doesn't look like a better UX as gRPC users get to supply this `int` value for filters. For CLI queries, we are taking `string` as the request input and converting it back to `int`. Considering that as a better UX, we thought of introducing the same in gRPC queries as well. 
We had two solutions,
1) Introduce an interceptor/middleware for gRPC to handle this conversion
2) Change gRPC query logic directly to handle it

As CLI is also using gRPC's `queryClient` for queries, adding an interceptor just for gRPC or gRPC gateway requests didn't sound like a plan. We discussed internally and thought changing gRPC filters to consider `string` as input for filters and converting them back to original types as needed. With this, all the clients (CLI, gRPC, gRPC gateway) can use `string` for filters and all the casting logic takes place in gRPC queryClient.

We can replace this hard coded string value with some constant like `DEPLOYMENT_STATUS_CLOSED` or something, but this logic lgtm